services/motion/builtin/builtin_test.go

```diff
@@ -1026,3 +1027,128 @@ func TestPlanHistory(t *testing.T) {
+		}
+		_, err = ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("starting outside of bounding regions fails", func(t *testing.T) {
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, origin, nil, 5)
+		defer ms.Close(ctx)
+
+		box, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{2222, 2222, 2}, "")
+		test.That(t, err, test.ShouldBeNil)
+
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Destination:        dst,
+			BoundingRegions: []*spatialmath.GeoGeometry{
+				spatialmath.NewGeoGeometry(geo.NewPoint(20, 20), []spatialmath.Geometry{box}),
+			},
+			MotionCfg: motionCfg,
+			Extra:     extra,
+		}
+		_, err = ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldNotBeNil)
+		expectedErrorString := "frame named test-base is not within the provided bounding regions"
+		test.That(t, strings.Contains(err.Error(), expectedErrorString), test.ShouldBeTrue)
+	})
+
+	t.Run("implicit success with no bounding regions", func(t *testing.T) {
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, origin, nil, 5)
+		defer ms.Close(ctx)
+
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Destination:        dst,
+			MotionCfg:          motionCfg,
+			Extra:              extra,
+		}
+		_, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("fail to plan outside of bounding regions", func(t *testing.T) {
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, origin, nil, 5)
+		defer ms.Close(ctx)
+
+		box, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{500, 500, 2}, "")
+		test.That(t, err, test.ShouldBeNil)
+
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Destination:        dst,
+			MotionCfg:          motionCfg,
+			BoundingRegions: []*spatialmath.GeoGeometry{
+				spatialmath.NewGeoGeometry(geo.NewPoint(0, 0), []spatialmath.Geometry{box}),
+			},
+			Extra: extra,
+		}
+		_, err = ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err, test.ShouldBeError, errors.New("destination was not within the provided bounding regions"))
+	})
+
+	t.Run("list of bounding regions - success case", func(t *testing.T) {
+		// string together multiple bounding regions, such that the robot must
+		// actually account for them in order to create a valid path
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, origin, nil, 5)
+		defer ms.Close(ctx)
+
+		box1, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{500, 500, 2}, "")
+		test.That(t, err, test.ShouldBeNil)
+		box2, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{555.5, 0, 0}),
```

Comment:
Could we alter this test slightly? As far as I can tell this second region also overlaps the start position; it does not test that a base can move into a region solely marked by a bounding region that it did not begin the motion in contact with.

Reply:
i'll update this.